src/diffusers/loaders.py

```diff
@@ -962,241 +1089,64 @@ def lora_scale(self) -> float:
+                raise ValueError(f"{text_encoder.__class__.__name__} does not support LoRA training")
+
+    @classmethod
+    def _modify_text_encoder(cls, text_encoder, lora_scale=1, network_alpha=None):
         r"""
         Monkey-patches the forward passes of attention modules of the text encoder.
-
-        Parameters:
-            attn_processors: Dict[str, `LoRAAttnProcessor`]:
-                A dictionary mapping the module names and their corresponding [`~LoRAAttnProcessor`].
         """
 
         # First, remove any monkey-patch that might have been applied before
-        self._remove_text_encoder_monkey_patch()
-
-        # Loop over the CLIPAttention module of text_encoder
-        for name, attn_module in self.text_encoder.named_modules():
-            if name.endswith(TEXT_ENCODER_ATTN_MODULE):
-                # Loop over the LoRA layers
-                for attn_proc_attr, text_encoder_attr in self._lora_attn_processor_attr_to_text_encoder_attr.items():
-                    # Retrieve the q/k/v/out projection of CLIPAttention and its corresponding LoRA layer.
-                    module = attn_module.get_submodule(text_encoder_attr)
-                    lora_layer = attn_processors[name].get_submodule(attn_proc_attr)
+        cls._remove_text_encoder_monkey_patch(text_encoder)
 
-                    # save old_forward to module that can be used to remove monkey-patch
-                    old_forward = module.old_forward = module.forward
+        lora_parameters = []
 
-                    # create a new scope that locks in the old_forward, lora_layer value for each new_forward function
-                    # for more detail, see https://github.com/huggingface/diffusers/pull/3490#issuecomment-1555059060
-                    def make_new_forward(old_forward, lora_layer):
-                        def new_forward(x):
-                            result = old_forward(x) + self.lora_scale * lora_layer(x)
-                            return result
+        for _, attn_module in text_encoder_attn_modules(text_encoder):
+            if isinstance(text_encoder, CLIPTextModel):
+                attn_module.q_proj = PatchedLoraProjection(attn_module.q_proj, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.q_proj.lora_linear_layer.parameters())
 
-                        return new_forward
-
-                    # Monkey-patch.
-                    module.forward = make_new_forward(old_forward, lora_layer)
-
-    @property
-    def _lora_attn_processor_attr_to_text_encoder_attr(self):
-        return {
-            "to_q_lora": "q_proj",
-            "to_k_lora": "k_proj",
-            "to_v_lora": "v_proj",
-            "to_out_lora": "out_proj",
-        }
+                attn_module.k_proj = PatchedLoraProjection(attn_module.k_proj, lora_scale, network_alpha)
+                lora_parameters.extend(attn_module.k_proj.lora_linear_layer.parameters())
 
-    def _load_text_encoder_attn_procs(
```

Comment:
We only ever call this function with a state dict to no need to have a separate loading procedure